web-console/src/utils/date-floor-shift-ceil/date-floor-shift-ceil.ts

```diff
@@ -0,0 +1,296 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { fromDate, startOfWeek } from '@internationalized/date';
+
+export type AlignFn = (dt: Date, tz: string) => Date;
+
+export type ShiftFn = (dt: Date, tz: string, step: number) => Date;
+
+export type RoundFn = (dt: Date, roundTo: number, tz: string) => Date;
+
+export interface TimeShifterNoCeil {
+  canonicalLength: number;
```

Comment:
nit: should it be more obvious that this number is millis?

Reply:
in general in JS any number time related is assumed to be in ms unless stated otherwise